PerfTools/AllocMonitor/scripts/edmModuleAllocJsonToCircles.py

```diff
@@ -38,15 +38,46 @@ def processModuleTransition(moduleLabel, moduleType, moduleInfo, transitionType,
     Any missing field defaults to 0.
 
     Note: Entries with record names are excluded as they belong to EventSetup transition only.
+    
+    For the "source" module and "event" transition, sum all alloc records with the same 
+    run, lumi, and event number.
     """
     moduleKey = UniqueKey(moduleLabel, moduleType, "")
     moduleTransition[moduleKey] = {"cpptype": moduleType, "allocs": []}
-    for entry in moduleInfo:
-        # Only process entries that match the transition type AND don't have record names
-        # (entries with record names are EventSetup only)
-        if (entry.get("transition", None) == transitionType and
-            not ("record" in entry and "name" in entry["record"])):
-            moduleTransition[moduleKey]["allocs"].append(entry.get("alloc", {}))
+    
+    # Special handling for "source" module with "event" transition
+    if moduleLabel == "source" and transitionType == "event":
+        # Group entries by (run, lumi, event)
+        event_groups = {}
+        for entry in moduleInfo:
+            if (entry.get("transition", None) == transitionType and
+                not ("record" in entry and "name" in entry["record"])):
+                sync = entry.get("sync", {})
+                key = (sync.get("run", 0), sync.get("lumi", 0), sync.get("event", 0))
+                if key not in event_groups:
+                    event_groups[key] = []
+                event_groups[key].append(entry.get("alloc", {}))
```

Comment:
I believe in principle you could do the summation already here, that would avoid looping through events and the 5 loops over allocation records of Source per event below (and the memory of `event_groups`).

If the script is still fast to run, this comment is not very relevant.